### PR TITLE
feat: Update or delete member

### LINKED ISSUE #185: Update or delete member
### Problem
Currently you cannot change member's role or remove them from the project.

### Solution
* Run the seed file `yarn db:seed`
* Go to `projects/:id/members` page
* Select active tab
* Click on the settings icon
* Open a modal with select input (current role selected by default)
* Have two buttons, one to remove member another one to save and close the modal.
* Allow only `owner` and `maintainer` to make this change.
* When you click on remove member, do not delete the member, but set `access.active` to `false`

### PR DESCRIPTION
**Pull request by ttebify**
# Description

> Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #185

## Type of change

> Please delete options that are not relevant.

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

> Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

1. Logged in as a project owner, I tried to update the role of another user to owner. I verified that the update was not allowed and the appropriate error message was displayed.
2. Logged in as a maintainer, I tried to update the role of an owner. I verified that the update was not allowed and the appropriate error message was displayed.
3. Logged in as a maintainer, I tried to update my own role. I verified that the update was not allowed and the appropriate error message was displayed.
4. Logged in as a member with no special privileges, I tried to update the role of another member. I verified that the update was not allowed and the appropriate error message was displayed.

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I documented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: prisma/schema.prisma [hunk 1]
```diff
@@ -94,7 +103,7 @@ model Access {
   updatedAt DateTime  @updatedAt
   userId    String
   projectId String
-  role      String?   @default("owner")
+  role      UserRole
```
Comment (dahal) [inline prisma/schema.prisma:106]: There is a migration missing for this default change

`ALTER TABLE "Access" ALTER COLUMN "role" DROP DEFAULT;`

If you reset your database and try running the migration/seed from scratch. You will notice.

Comment (ttebify) [reply prisma/schema.prisma:106]: The newly generated column was dropped entirely and recreated, if you check on the migrations folder you'd see

File: prisma/migrations/20230319022555_make_role_field_required_and_change_type_to_enum_in_access_table/migration.sql [hunk 2]
```diff
@@ -0,0 +1,17 @@
+/*
+  Warnings:
+
+  - The `role` column on the `Access` table would be dropped and recreated. This will lead to data loss if there is data in the column.
+  - The `role` column on the `ProjectInvite` table would be dropped and recreated. This will lead to data loss if there is data in the column.
+
+*/
+-- CreateEnum
+CREATE TYPE "UserRole" AS ENUM ('maintainer', 'owner', 'editor', 'viewer', 'developer', 'guest');
```
Comment (dahal) [inline prisma/migrations/20230319022555_make_role_field_required_and_change_type_to_enum_in_access_table/migration.sql:9]: Let's limit roles to `'maintainer', 'owner', 'developer', 'guest'` for now. Down the line we will have scope based custom roles. It becomes harder to support multiple roles without scopes.

File: prisma/schema.prisma (hunk 1 above)
Comment (ttebify) [reply prisma/schema.prisma:106]: Ok I see it now, I'll work on that

**Review by dahal — APPROVED** [review]
Thanks @ttebify , awesome job. 🚀 
Also, this has already been merged with #279 if you have additional changes, please create a new PR.

**Comment by ttebify** [comment]
> Thanks @ttebify , awesome job. rocket Also, this has already been merged with #279 if you have additional changes, please create a new PR.

Awesome 🥳, thank you for taking the time it.

**Pull request closed without merging**